include/nbl/builtin/hlsl/subgroup/arithmetic_portability_impl.hlsl

```diff
@@ -330,30 +330,30 @@ struct exclusive_scan
 
     T operator()(T value)
     {
-		const uint subgroupInvocation = glsl::gl_SubgroupInvocationID();
-		
-		value = impl(value);
-		// store value to smem so we can shuffle it
-		uint left = glsl::subgroupShuffleUp<T>(value, 1);
-		value = subgroupInvocation >= 1 ? left : Binop::identity(); // the first invocation doesn't have anything in its left so we set to the binop's identity value for exlusive scan
-		// return it
-		return value;
+        const uint subgroupInvocation = glsl::gl_SubgroupInvocationID();
+        
+        value = impl(value);
+        // store value to smem so we can shuffle it
+        uint left = glsl::subgroupShuffleUp<T>(value, 1);
+        value = subgroupInvocation >= 1 ? left : Binop::identity(); // the first invocation doesn't have anything in its left so we set to the binop's identity value for exlusive scan
+        // return it
+        return value;
     }
 
 // protected:
-	inclusive_scan<T, Binop> impl;
+    inclusive_scan<T, Binop> impl;
 };
 
 template<typename T, class Binop>
 struct reduction
 {
     T operator()(T value)
     {
-		value = impl(value);
-		value = glsl::subgroupBroadcast<T>(value, LastSubgroupInvocation()); // take the last subgroup invocation's value for the reduction
-		return value;
+        value = impl(value);
+        value = glsl::subgroupBroadcast<T>(value, LastSubgroupInvocation()); // take the last subgroup invocation's value for the reduction
+        return value;
     }
-	
+    
 // protected:
     inclusive_scan<T, Binop> impl;
```

Comment:
1. is there any need to keep these around as members (as opposed to conjuring them up as temporaries inside the function)?
2. why not make these into functions instead of structs? there's no accessors or partial specialization necesary right?

Reply:
need (2)